src/amperity/gocd/agent/aurora/agent.clj

```diff
@@ -87,6 +93,11 @@
      (when-not (re-matches #"[a-z]+" (:agent_tag settings))
        {:key :agent_tag
         :message "Agent tag must consist of lowercase letters"})
+     (when-let [environments (not-empty (:environments settings))]
+       (when-not (re-matches #"[a-zA-Z0-9_\-]{1}[a-zA-Z0-9_\-.]*(,[a-zA-Z0-9_\-]{1}[a-zA-Z0-9_\-.]*)*"
```

Comment:
Isn't `[a-z]{1}` the same as just `[a-z]`?
```suggestion
       (when-not (re-matches #"[a-zA-Z0-9_\-][a-zA-Z0-9_\-.]*(,[a-zA-Z0-9_\-][a-zA-Z0-9_\-.]*)*"
```
Alternately, this might be simpler if you try to parse and then validate the environment names individually:
```clojure
(when-not (str/blank? (:environments settings))
  (when-let [bad-envs (->>
                        (str/split (:environments settings) #"\s*,\s*")
                        (remove (partial re-matches #"[a-zA-Z0-9_-][a-zA-Z0-9_.-]*"))
                        (seq))]
    {:key :environments
     :message (format "..." ,,,)}))
```